app/controllers/siamese_bert_similarity.py

```diff
@@ -34,39 +30,43 @@ def post(self):
 
 class SiameseBertSimilarity(Resource):
 
-    def __init__(self):
-        super()
-
-        # general BERT model related things
-        vocab_file = os.path.join(BERT_PRETRAINED_DIR, 'vocab.txt')
-        config_file = os.path.join(BERT_PRETRAINED_DIR, 'bert_config.json')
-        do_lower_case = BERT_MODEL.startswith('uncased')
-
-        # checkpoint to initialize from
-        checkpoint_path = tf.train.latest_checkpoint(TRAINED_MODEL_DIR)
-        # checkpoint_path = os.path.join(BERT_PRETRAINED_DIR, 'bert_model.ckpt')
-
-        # featurization related tools
-        self.processor = FtmProcessor()
-        self.label_list = self.processor.get_labels()
-        self.tokenizer = tokenization.FullTokenizer(
-            vocab_file=vocab_file,
-            do_lower_case=do_lower_case)
-
-        # estimator setup
-        run_config = tf.estimator.RunConfig()
-        model_fn = self._model_fn_builder(
-            bert_config=modeling.BertConfig.from_json_file(config_file),
-            init_checkpoint=checkpoint_path)
-        self.estimator = tf.estimator.Estimator(
-            model_fn=model_fn,
-            config=run_config)
-
-        # set up parser
-        self.post_parser = reqparse.RequestParser()
-        self.post_parser.add_argument("doc1", action='append', required=True)
-        self.post_parser.add_argument("doc2", action='append', required=True)
-        self.post_parser.add_argument("sort", required=False)
+    model_save_dirs = [_dir for _dir in Path(SAVE_MODEL_BASE_DIR).iterdir()
+                       if _dir.is_dir() and 'temp' not in str(_dir)]
+    model_save_dir = str(sorted(model_save_dirs)[-1])
+    # model_save_dir = SAVE_MODEL_BASE_DIR + '/optimized'
+    print(f'--- Loading the model saved at: {model_save_dir}')
+
+    predict_fn = predictor.from_saved_model(model_save_dir)
+    print(f'--- Done loading the model!')
+
+    print('--- computing dummy input... ---')
+    # run in dummy input since model has some overhead on the first prediction
+    dummy_val = predict_fn({
+        'l_input_ids': [[0] * model_configs.MAX_SEQ_LENGTH],
+        'r_input_ids': [[0] * model_configs.MAX_SEQ_LENGTH],
+        'l_input_mask': [[0] * model_configs.MAX_SEQ_LENGTH],
+        'r_input_mask': [[0] * model_configs.MAX_SEQ_LENGTH]})
+    print('--- done computing dummy input... ---')
```

Comment:
instead of print using loggers would be ideal

Reply:
better to do in this PR or in a separate PR?

Reply:
talked offline, can do in a separate PR